react-web/src/compositions/Form/components/Input.tsx

```diff
@@ -5,53 +5,66 @@ import "./Input.scss";
 import { useFormContext } from "react-hook-form";
 
 interface InputProps extends ComponentProps<"input"> {
-  label: string;
-  disabled?: boolean;
+ label: string;
+ disabled?: boolean;
+ disablefocus?: boolean;
 }
 
 export const Input = forwardRef<HTMLInputElement, InputProps>(function Input(
-  { 
-    disabled = false,
-    label,
-    type = "text",
-    className = "",
-    name,
-    ...props 
-  },
-  ref
+ {
+   disabled = false,
+   label,
+   type = "text",
+   className = "",
+   disablefocus = false,
+   name,
+   ...props
+ },
+ ref
 ) {
-  const {
-    formState: { errors },
-  } = useFormContext();
-
-  const [active, setActive] = useState(false);
-  const [errorMsg, setErrorMessage] = useState("");
-
-  // eslint-disable-next-line
-  useEffect(() => {
-    if (errors && name && errors[name]) {
-      setErrorMessage(errors[name]?.message as string);
-    } else setErrorMessage("");
-  });
-
-  return (
-    <div
-      className={`input-wrapper ${className}`}
-      onBlur={(e: any) => !e.target.value && setActive(false)}
-      onClick={(e: any) => setActive(true)}
-    >
-      <div
-        className={`input ${active ? "active" : ""} ${errorMsg ? "error" : ""} ${disabled ? "disabled" : ""}`}
-        onClick={(_) => {
-          setActive(true);
-          document.getElementById(name || "")?.focus();
-        }}
-      >
-        <label>{label}</label>
-        <input type={type} ref={ref} {...props} name={name} id={name} data-testid={name} />
-      </div>
-
-      {errorMsg && <FieldError message={errorMsg} />}
-    </div>
-  );
+ const {
+   formState: { errors },
+ } = useFormContext();
+
+ useEffect(() => {
+   disablefocus && setActive(true);
+ }, [disablefocus]);
+
+ const [active, setActive] = useState(false);
+ const [errorMsg, setErrorMessage] = useState("");
+
+ // eslint-disable-next-line
+ useEffect(() => {
+   if (errors && name && errors[name]) {
+     setErrorMessage(errors[name]?.message as string);
+   } else setErrorMessage("");
+ });
```

Comment:
Upon providing `[errors]`, the form error messages weren't being displayed as expected. Hence, removed it as a bug fix.

Reply:
Indeed doing that will fix apparently your issue. But it's not a good practice in React to change a given state at every arbitrary render just to cover all the possibilities. 
What you have to do it's to show a message every time the message it's changing, not call the show function every time in the hope it will cover also the margin case when an error appears or changes

I guess the issue it's you don't provide all the dependencies: `errors`, `name`, `errors[name]?.messages` 

this is  a suggestion  ( caution I didn't run it, or check it against TS)

```javascript
 const  lastError = (errors && (errors[name]?.message as string)) || "";
 useEffect(() => setErrorMessage(lastError),[lastError]);
```

Reply:
Have re-written this, removing unwanted state updates on every re-render. Please have a look now